Implementation of app state (#189)
pupilfirst/wd301-scripts: react-app-state/implementing-app-state/README.md

# Script
In this lesson we will implement our application state, so let's get started.

### Step 1: Preparing the reducer
So, to start with the implementation, first we will move the `reducer` function from the `ProjectList` component to the `src/context/projects/reducer.ts` file.
```ts
interface Project {
  id: number;
  name: string;
}

> // Dialogue 1: I'll rename this interface from State to ProjectsState. And I'll also export the interface.

Reviewer: ```suggestion
// Dialogue 1: I'll rename the interface in the `ProjectList` component from `State` to `ProjectsState`. And I'll also export the interface.
```

                        Cancel
                      </button>
                    </form>
                  </div>
                </Dialog.Panel>
              </Transition.Child>
            </div>
          </div>
        </Dialog>
      </Transition>    
    </>
  )
}

export default NewProject;
```

Alright! finally we're ready to test the enrire projects module. Let's do it.
> Open http://localhost:3000 in browser and create a project.

So, as you can see, after creating a project, the new project is showing up in the projects list, automatically. Right? 

This is what we intended to build since the beginning, and finally we've achieved it, with the help of application-level state.

# Text
So, finally we've copmpleted the implementation of projects module and it's working as expected. In this level, we've learned a lot of new concepts like:
- What is context?
- What is useReducer?
- How to use context and useReducer together to design app-level state? etc.

> We've written a lot of code and we've also structured the code well. We've spearated different concerns in differend subfolders inside the `src` directory. In that process, lot of old code became absolete. It's time to remove them. Let's do it.

Reviewer: ```suggestion
We've written a lot of code and we've also structured the code well. We've separated different concerns in differend subfolders inside the `src` directory. In that process, lot of old code became absolete. It's time to remove them. Let's do it.
```
